### PR TITLE
[AIRFLOW-40] add DAGs' filtering over ldap groups

### PR DESCRIPTION
**Pull request by dsjl**
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- https://issues.apache.org/jira/browse/AIRFLOW-40

### CODE REVIEW

File: airflow/www/views.py [hunk 1]
```diff
@@ -1725,14 +1725,38 @@ def index(self):
         qry = None
         # filter the dags if filter_by_owner and current user is not superuser
         do_filter = FILTER_BY_OWNER and (not current_user.is_superuser())
+        owner_mode = conf.get('webserver', 'OWNER_MODE').strip().lower()
+
+        if do_filter and owner_mode not in ['user', 'ldapgroup']:
+            logout_user()
+            flash("Configuration error: "
```
Comment (bolkedebruin) [inline airflow/www/views.py]: I think it is better to move this to configuration._validate

Comment (dsjl) [inline airflow/www/views.py]: Would adding an 'elif' in ConfigParserWithDefaults._validate be sufficient or do you know if a better config validation will happen in the near future?

**Comment by bolkedebruin** [comment]
Already better. Can you extend the tests to cover this? And make sure to squash your commits.

File: airflow/www/views.py [hunk 2]
```diff
@@ -1725,14 +1725,25 @@ def index(self):
...
+
         if do_filter:
-            qry = (
-                session.query(DM)
-                    .filter(
-                    ~DM.is_subdag, DM.is_active,
-                    DM.owners == current_user.username)
-                    .all()
-            )
+            if owner_mode == 'ldapgroup':
```
Comment (bolkedebruin) [inline airflow/www/views.py]: This can be refactored to be a bit cleaner and not repeat code too much. qry = qry.filter()

**Comment by bolkedebruin** [comment]
Please rebase and squash your commits. Don't forget to adhere to the commit guidelines.

**Comment by dsjl** [comment]
Hi @bolkedebruin,

Do you see anything that can still be improved to have this PR merged into master?

Thanks for your help.

**Comment by dsjl** [comment]
@bolkedebruin , thanks a lot for your useful comments that definitely helped improving the code quality. If you do not have any other comments, this PR can be considered finished from my side.

Cheers

**Pull request closed without merging**